tests/src/test/java/org/teavm/classlib/java/util/stream/StreamTest.java

```diff
@@ -287,6 +287,27 @@ public void minMax() {
 
         assertFalse(Stream.<Integer>empty().min(Comparator.comparing(x -> x)).isPresent());
         assertFalse(Stream.<Integer>empty().max(Comparator.comparing(x -> x)).isPresent());
+
+        BrokenComparable first = new BrokenComparable(0, 0);
+        BrokenComparable second = new BrokenComparable(0, 1);
+
+        assertEquals(0, Stream.of(first, second).max(Comparator.comparing(x -> x)).get().id);
+        assertEquals(0, Stream.of(first, second).min(Comparator.comparing(x -> x)).get().id);
+    }
+
+    private static class BrokenComparable implements Comparable<BrokenComparable> {
```

Comment:
Does this class really need to be Comparable?

Reply:
Is "BrokenComparable" a right name? What about just IntPair or something like this?

Reply:
The reason of this name is because comparable isn't aligned with possible "equals", where equals will be by id, but compared values will be by value. So in similar cases (I had more complex), we are expecting max to return first found object, but in reality it returned last (JVM in both cases for Collections.max and Stream.max returns first).

Reply:
So, name means conscious break of "compare-equals" contract for some reason. Can rename it to IntPair.